Homework8: Ekaterina Zavadskaya

KateZavadskaya opened the pull request:
Feat(hm8): add homework8

Docs: hm8_1.py - hm8_3.py

KateZavadskaya committed:
Feat(hm8): add three epmty files for hm8

Docs: hm8_1.py - hm8_3.py

KateZavadskaya committed:
Feat(hm8): add hm8

Add done hm8

Docs: hm8_1.py - hm8_3.py

KateZavadskaya committed:
Fix(code): fix linter

Fix linter in hm8_3.py: flake8, Mypy, Pylint

KateZavadskaya committed:
Fix(linter): add linter fix at hm8_2.py

Doc: hm8_2.py

KateZavadskaya commented:
@Stml89 , success with homework8 💃, please

Stml89 commented on hm8_1.py:
```diff
@@ -0,0 +1,28 @@
+print(f"3 {solution([1, 2, 1, 2])}")
+print(f"4 {solution([1, 3, 2, 1])}")
+print(f"5 {solution([1, 2, 3, 4, 5, 3, 5, 6])}")
+print(f"6 {solution([40, 50, 60, 10, 20, 30])}")
```
better to use `assert`, instead of print() to check that it work as expected

KateZavadskaya replied:
done 👍, also according to this comment add changes to hm8_2.py, please

Stml89 replied:
Nice, that you have checked another file and corrected issues there as well! Well done!

Stml89 replied:
Also, it is better if you have more tests to your code:
```
assert solution([1, 3, 2]) == ..., ...
...
assert solution([1, 2, 3, 4, 5, 3, 5, 6]) == ..., ...
...
assert solution([40, 50, 60, 10, 20, 30] == ..., ...
```

KateZavadskaya replied:
I will take it into account in the next homework, thank you!🌸

Stml89 commented on hm8_3.py:
```diff
@@ -0,0 +1,33 @@
+                        sum_even += doubled
+            return (sum_even + sum_odd) % 10 == 0
+        break
+    return None
```
return False

KateZavadskaya replied:
done 👍, also according to this comment add changes to hm8_2.py, please

KateZavadskaya committed:
Style(code): add improvements according comments

hm8_1 -  use assert, instead of print()
hm8_3 - change None to False + bonus: add assert instead of print
bonus hm8_2 - change return to return *discroption*

Docs: hm8_1.py - hm8_3.py
